src/application/service_tester.go

```diff
@@ -16,12 +17,42 @@ type ServiceTester struct {
 	BuiltDependencies map[string]config.AppDevelopmentDependency
 	AppDir            string
 	ServiceDir        string
-	*Initializer
-	*Runner
+	Initializer       *Initializer
+	Runner            *Runner
 }
 
 // NewServiceTester is ServiceTester's constructor
-func NewServiceTester(role string, serviceConfig types.ServiceConfig, builtDependencies map[string]config.AppDevelopmentDependency, appDir, serviceDir string, initializer *Initializer, runner *Runner) (*ServiceTester, error) {
+func NewServiceTester(serviceContext types.ServiceContext, logger *util.Logger, mode composebuilder.BuildMode) (*ServiceTester, error) {
+	appConfig := serviceContext.AppContext.Config
+	serviceConfig := serviceContext.Config
+	serviceDir := serviceContext.Location
+	role := serviceContext.Name
+	appDir := serviceContext.AppContext.Location
+	dockerComposeProjectName := fmt.Sprintf("%stests", composebuilder.GetDockerComposeProjectName(appDir))
```

Comment:
Can we still get this from a shared function like `getTestDockerComposeProjectName`? I think `exo clean` may need it.

Reply:
Yeah I haven't cleaned up the service tester yet. Ideally I don't want anything in application to know about docker compose stuff. The docker package should actually take care of this IMO